src/utils.ts

```diff
@@ -120,12 +130,16 @@ export function getConfig(
     tdmlConfig: tdmlConfig
       ? tdmlConfig
       : {
-          action: defaultConf.get('tdmlAction', 'none'),
-          name: defaultConf.get('tdmlName', '${command:AskforTDMLName}'),
-          description: defaultConf.get(
-            'tdmlDescription',
-            '${command:AskForTDMLDescription}'
-          ),
+          action: tdmlAction,
+          name: tdmlPropsNeeded
+            ? defaultConf.get('tdmlName', '${command:AskforTDMLName}')
+            : defaultConf.get('tdmlName', 'tdmlNamePlaceholder}'),
```

Comment:
Hmmm. Why create a different text placeholder string? 

Also, what is a tdmlName - there are too many kinds of names - the TDML file's name, the testSuite name, the test itself name,... so use a longer more explicit identifier. 

Watch out also for "config", as there is a testSuite defaultConfig and each test has a config also, and a testSuite can have multiple named defineConfig children. 

Suggest name "tdmlVSCodeConfig" would keep things clearer. 

Ideally, this should prompt you for a TDML file name, and then store it here, i.e., be sticky about the selection.

Reply:
> Hmmm. Why create a different text placeholder string?
> 
The root issue of #622 is that VSCode is trying to resolve ${command:AskforTDMLName} even in cases where it shouldn't, i.e. when a user runs a launch config with TDMLAction set to 'none'. The idea of this placeholder is to make it so that configs that are created won't use the command as the default value, which would require users to modify the default launch config to make it work for the default use case. On the other hand, if we run into a config with a TDMLAction that isn't none, we want the command to be the default value.

At best, this only reduces user impact for #622, so we're still looking into ways to fix it.

Thanks for the other suggestions - I don't want to try to overcrowd 1.3.0, but these will be helpful when we look to improve upon the TDML functionality in upcoming releases.

Reply:
I don't think there should be an action of type `none`; if you don't want a TDML action then the TDML config section should be optional. Then there won't be any need to suppress variable interpolations, and if you do want an action then the variables will get interpolated.